src/ray/util/logging.h

```diff
@@ -87,6 +88,18 @@ enum class RayLogLevel {
 
 #endif  // NDEBUG
 
+/// Macros for RAY_LOG_EVERY_N
+#define RAY_LOG_EVERY_N_VARNAME(base, line) RAY_LOG_EVERY_N_VARNAME_CONCAT(base, line)
+#define RAY_LOG_EVERY_N_VARNAME_CONCAT(base, line) base##line
+
+#define RAY_LOG_OCCURRENCES RAY_LOG_EVERY_N_VARNAME(occurrences_, __LINE__)
+
+#define RAY_LOG_EVERY_N(level, n)                             \
+  static std::atomic<uint64_t> RAY_LOG_OCCURRENCES(0);        \
+  if (ray::RayLog::IsLevelEnabled(ray::RayLogLevel::level) && \
```

Comment:
It looks good, thou I'm wondering if we can delegate to glog under the hood which presumably already supports this

Reply:
yup. this basically copied over what GLOG is doing: https://github.com/google/glog/blob/master/src/glog/logging.h.in#L985